src/middleware/router.coffee

```diff
@@ -166,6 +183,7 @@ buildNonPrimarySendRequestPromise = (ctx, route, options, path) ->
 
     ctx.routes = [] if not ctx.routes
     ctx.routes.push routeObj
```

Comment:
@debonair do we still need to attach the `routeObj` to the `ctx` with this async processing. Maybe we can rather just use the route object directly. I think when you return the `routeObj` it will become available as in the `.then` function here: https://github.com/jembi/openhim-core-js/pull/491/files#diff-ffc48656de2a6c6b5ee8a785b33109b0R153

Chat to me more about this if you like, I'm if what I've said above is exactly how it works, but I think it is.